File: src/sqlmap/SqlmapSession.java
package sqlmap;

import java.io.*;
import java.net.*;


/**
 * SqlmapSession class
 * 
 * Contains the session information:
 *     _host [no default]
 *     _port [8775]
 * Processes the api request:
 *     get()
 *     post()
 */
public class SqlmapSession {
	private String _host = null;
	private int _port = 8775; // default
	
	/**
	 * Create the SqlmapSession object
	 * 
	 * @param  host
	 *         string of the hostname
	 * @param  port
	 *         int of port number
	 */
	public SqlmapSession(String host, int port) {
		_host = host;
		_port = port;
	}
	public SqlmapSession(String host) {
		_host = host;
		_port = 8775;
	}
	
	/**
	 * Execute an HTTP Get request and return a string result
	 * 
	 * @param  page
	 *         string of the task page
	 * @return a string containing the request result
	 */
	public String executeGet(String page) {
		try {
			StringBuilder result = new StringBuilder();
			URL url = new URL("http://" + _host + ":" + _port + page);
			HttpURLConnection conn = (HttpURLConnection) url.openConnection();
			conn.setRequestMethod("GET");
			BufferedReader br = new BufferedReader(new InputStreamReader(conn.getInputStream()));
			String line;
			while ((line = br.readLine()) != null) {
				result.append(line);
			}
			br.close();
			return result.toString();
		} catch(Exception ex) {
			ex.printStackTrace();
			return null;
		}
	}
	
	/**
	 * Execute an HTTP Post request and return a string result
	 * 
	 * @param  page
	 *         string of the task page
	 * @param  data
	 *         string of post parameters data String
	 * @return a string containing the request result
	 */
	public String executePost(String page, String data) {
		try {
			StringBuilder result = new StringBuilder();
			URL url = new URL("http://" + _host + ":" + _port + page);
			HttpURLConnection conn = (HttpURLConnection) url.openConnection();
			conn.setRequestMethod("POST");
			conn.setRequestProperty("Content-Type", "application/json");
			
			conn.setDoOutput(true);
			DataOutputStream wr = new DataOutputStream(conn.getOutputStream());
			wr.writeBytes(data);
			wr.flush();
			wr.close();
			
			BufferedReader br = new BufferedReader(new InputStreamReader(conn.getInputStream()));
			String line;
			while ((line = br.readLine()) != null) {
				result.append(line);
			}
			br.close();
			return result.toString();
		} catch(Exception ex) {
			ex.printStackTrace();
			return null;
		}
	}
}

File: SqlmapSession.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import requests

class SqlmapStatus(object):
    _status = None
    _retcode = -1

class SqlmapLogItem(object):
    _message = ''
    _level = ''
    _time = ''

class SqlmapSession(object):
    _host = None
    _port = 8775  # default

    def __init__(self, host, port=8775):
        self._host = host
        self._port = port
    
    def get(self, page):
        '''Execute an HTTP Get request and return a string result.
           `page` string link of the task.'''
        try:
            url = f'http://{self._host}:{self._port}{page}'
            r = requests.get(url)
            return r.json()
        except Exception as ex:
            print(ex)
            return None

    def post(self, page, data):
        '''Execute an HTTP Post request and return a string
        `page` string link of the task.
        `data` post params.'''
        try:
            url = f'http://{self._host}:{self._port}{page}'
            r = requests.post(url, json=data)
            return r.json()
        except Exception as ex:
            print(ex)
            return None